pricegraph/src/orderbook.rs

```diff
@@ -425,12 +428,21 @@ fn format_path(path: &[NodeIndex]) -> String {
         .join("->")
 }
 
-/// Returns true if an auction element is considered a dust order.
-fn is_dust_order(element: &Element) -> bool {
+/// Returns true if an auction element should be included in the price graph.
+///
+/// Currently auction elements are ignored if:
+/// - They are "dust" orders, that is their remaining amount or balance is less
+///   than the minimum amount that the exchange allows for trades
+/// - They have a `0` price numerator or denominator
+fn should_include_auction_element(element: &Element) -> bool {
     const MIN_AMOUNT_U128: u128 = MIN_AMOUNT as _;
     const MIN_AMOUNT_U256: U256 = U256([MIN_AMOUNT as _, 0, 0, 0]);
 
-    element.remaining_sell_amount < MIN_AMOUNT_U128 || element.balance < MIN_AMOUNT_U256
+    let is_dust_order =
+        || element.remaining_sell_amount < MIN_AMOUNT_U128 || element.balance < MIN_AMOUNT_U256;
+    let has_valid_price = || element.price.numerator != 0 && element.price.denominator != 0;
```

Comment:
```suggestion
    let has_valid_price = element.price.numerator != 0 && element.price.denominator != 0;
```

Reply:
I tried to be cleaver and use closures to take advantage of short cutting but that doesn't really make things clearer. I'll remove it.

Reply:
I missed the closure and function call below. I thought you had copy pasted this from the statement above. In this case the operation is so cheap that I also prefer the non closure version.

Reply:
Just checked in play ground and the compiler optimizes the let bindings away so it makes no difference and the closures only add confusion.